db/compaction/compaction_job.cc

```diff
@@ -621,6 +641,16 @@ void CompactionJob::GenSubcompactionBoundaries() {
     }
   }
 
+  Slice output_split_user_key;
+  const InternalKey output_split_key = c->GetOutputSplitKey();
+  if (output_split_key.Valid()) {
+    output_split_user_key = ExtractUserKey(output_split_key.Encode());
+    bounds.emplace_back(output_split_key.Encode());
```

Comment:
Confusingly, `InternalKey::Encode()` does not make a copy. I think that makes it unsafe for use in `bounds`.

Reply:
Replace `output_split_key` in 648 with `c->GetOutputSplitKey()` should be fine?